### PR TITLE
Create additional Snowflake detections

### PR DESCRIPTION
**Pull request by bseb**
### Background

Add some new Snowflake detection queries based on the Snowflake security checklist update

### Changes

* Create 3 new scheduled queries for Snowflake

### Testing

* Unit + CI/CD

Closes https://app.asana.com/0/1201300875909824/1201306651300009/f

### CODE REVIEW

File: snowflake_queries/snowflake_key_user_password_login.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+Enabled: false
+ScheduledQueries:
+  - Query.Snowflake.KeyUserPasswordLogin
+Severity: Medium
+Tests:
+  -
+    Name: Key user logged in with Password
+    ExpectedResult: true
+    Log:
+      query_text: "testuser RSAKey Password"
```
Comment (lindsey-w) [inline snowflake_queries/snowflake_key_user_password_login.yml:16]: is this the expected output from the query? What shows in the title output from this test?

File: snowflake_queries/snowflake_scim_token_created.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+Enabled: false
+ScheduledQueries:
+  - Query.Snowflake.SCIMTokenCreated
+Severity: Info
+Tests:
+  -
+    Name: SCIM Token Created
+    ExpectedResult: true
+    Log:
+      query_text: "system xyz generate_scim_access_token"
```
Comment (lindsey-w) [inline snowflake_queries/snowflake_scim_token_created.yml:15]: same question: is this the expected output from the query?

File: snowflake_queries/snowflake_key_user_password_login_query.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  --return instances where a user who has key-based login configured logs in with a password
+  --this was adapted from a Security Feature Checklist query
+
+   SELECT
+     u.name,
+     first_authentication_factor,
+     second_authentication_factor,
+     count(*)
+  FROM snowflake.account_usage.login_history as l
+  JOIN snowflake.account_usage.users u on l.user_name = u.name and has_rsa_public_key = 'true'
```
Comment (lindsey-w) [inline snowflake_queries/snowflake_key_user_password_login_query.yml:19]: is this searching all time? should we scope it to look back at the same timeframe as the schedule? (24 hours?)

File: snowflake_queries/snowflake_unusual_logins.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+AnalysisType: scheduled_rule
+Filename: snowflake_unusual_logins.py
+RuleID: Snowflake.UnusualLogins
```
Comment (lindsey-w) [inline snowflake_queries/snowflake_unusual_logins.yml:3]: This makes me think of the standard "Unusual login" detection. Wonder if we could rename to something else? 🤷🏻‍♀️

**Review by lindsey-w — COMMENTED** [review]
Nice work! love that we are expanding our out of the box snowflake detections

File: snowflake_queries/snowflake_key_user_password_login.yml (hunk 1 above)
Comment (bseb) [reply snowflake_queries/snowflake_key_user_password_login.yml]: Changed based on our conversation

File: snowflake_queries/snowflake_scim_token_created.yml (hunk 2 above)
Comment (bseb) [reply snowflake_queries/snowflake_scim_token_created.yml]: Changed based on our conversation

File: snowflake_queries/snowflake_unusual_logins.yml (hunk 4 above)
Comment (bseb) [reply snowflake_queries/snowflake_unusual_logins.yml]: Yeah thats a good call, I changed it to "Unusual login volume"

File: snowflake_queries/snowflake_unusual_login_volume_query.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+Description: >
+  Alert on a high number of logins (above a baselined threshold)
+AthenaQuery: >
+  /* athena query not supported */
+  SELECT count(1)
+SnowflakeQuery: >
+
+  SELECT DISTINCT user_name, count(user_name), first_authentication_factor
+  FROM snowflake.account_usage.login_history
+  WHERE p_occurs_since('10 h' ,, event_timestamp)
```
Comment (lindsey-w) [inline snowflake_queries/snowflake_unusual_login_volume_query.yml:13]: looks like the rate minutes specify this should run 1/day, but we only look at the last 10 hours of activity here.

**Review by lindsey-w — DISMISSED** [review]
Looks good! Just had a question about the rate minutes on one.

File: snowflake_queries/snowflake_unusual_login_volume_query.yml (hunk 5 above)
Comment (bseb) [reply snowflake_queries/snowflake_unusual_login_volume_query.yml]: Thanks for catching that! Remnant of me testing the query then pasting it back into the YAML

**Review by lindsey-w — APPROVED** [review]

**Pull request merged**